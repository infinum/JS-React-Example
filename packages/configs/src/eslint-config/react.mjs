import { fixupPluginRules } from '@eslint/compat';
import pluginReact from 'eslint-plugin-react';
import pluginReactHooks from 'eslint-plugin-react-hooks';

export default [
	{
		files: ['**/*.tsx', '**/*.jsx'],
		settings: {
			react: {
				version: 'detect',
			},
		},
		plugins: {
			react: pluginReact,
			'react-hooks': fixupPluginRules(pluginReactHooks),
		},
		rules: {
			...pluginReact.configs['jsx-runtime'].rules,
			...pluginReactHooks.configs.recommended.rules,
		},
	},
];
