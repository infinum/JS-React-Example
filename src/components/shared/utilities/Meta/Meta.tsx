import React, { FC } from 'react';
import Head from 'next/head';

interface IHeadProps {
	title?: string;
	description?: string;
}

export const Meta: FC<IHeadProps> = ({
	title = 'React Example | Infinum',
	description = 'Javascript React Example project landing page ',
}) => {
	return (
		<Head>
			<title>{title}</title>
			<meta id="meta-description" name="description" content={description} />
			<link rel="icon" href="/favicon.ico" />
		</Head>
	);
};
