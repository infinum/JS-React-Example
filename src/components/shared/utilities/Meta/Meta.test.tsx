import { ReactNode } from 'react';
import { render } from 'test-utils';
import { Meta } from './Meta';

jest.mock('next/head', () => {
	return {
		__esModule: true,
		default: ({ children }: { children: ReactNode }) => {
			return <>{children}</>;
		},
	};
});

// eslint-disable-next-line @typescript-eslint/no-unsafe-return
jest.mock('@chakra-ui/react', () => ({
	...jest.requireActual('@chakra-ui/react'),
	ChakraProvider: ({ children }: { children: ReactNode }) => {
		return <>{children}</>;
	},
}));

describe('Meta', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<Meta />, {
			container: document.head,
		});

		expect(asFragment()).toMatchSnapshot();
	});

	it('should match snapshot with custom title and description', () => {
		const { asFragment } = render(<Meta title="Custom title" description="Custom description" />, {
			container: document.head,
		});

		expect(asFragment()).toMatchSnapshot('custom-title-and-description');
	});
});
