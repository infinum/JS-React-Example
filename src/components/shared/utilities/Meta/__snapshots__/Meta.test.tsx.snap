// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta should match snapshot with custom title and description: custom-title-and-description 1`] = `
<DocumentFragment>
  <title>
    Custom title
  </title>
  <meta
    content="Custom description"
    id="meta-description"
    name="description"
  />
  <link
    href="/favicon.ico"
    rel="icon"
  />
</DocumentFragment>
`;

exports[`Meta should matches snapshot 1`] = `
<DocumentFragment>
  <title>
    React Example | Infinum
  </title>
  <meta
    content="Javascript React Example project landing page "
    id="meta-description"
    name="description"
  />
  <link
    href="/favicon.ico"
    rel="icon"
  />
</DocumentFragment>
`;
